Dockerfile

```diff
@@ -0,0 +1,39 @@
+
+from node:lts as builder
+workdir hubs
+copy . .
```

Comment:
yeah i was thinking to run in only in github action's hosted agent so caching's not applicable there, we do have option to use github action's caching feature but i'm not yet sure if the few minutes saved would worth the cost of extra pipeline complexity + losing the portability of wrapping everything inside dockerfile
I think for local testing a custom Dockerfile should be used, and depends on the individual developer it could be different, maybe we can later add an example in the readme for it?

Reply:
Ah yeah, I assumed github actions just had built in handling for this... 

Some quick searching shows it shouldn't be too hard to setup https://github.com/progapandist/anycable_rails_demo/blob/master/.github/workflows/fake_deploy_multistage.yml (from https://evilmartians.com/chronicles/build-images-on-github-actions-with-docker-layer-caching) by using github's cache action to cache the build cache directory. 

Re  "extra pipeline complexity + losing the portability of wrapping everything inside dockerfile" I don't think this moves anything out of the dockerfile, just configures the environment it is running in. Not the highest priority ever, but shaving off build time definitely adds up. Our current CI builds take quite a long time, and its highly dominated by `npm ci` time which actually changes quite rarely. Even if we don't go through setting up github actions to enable caching, making this part of the dockerfile more cache friendly seems good (unless there is some tradeoff)

Reply:
yup agreed i'll add the github action caching -- just discovered yesterday that file ops are like 4-6 times slower in "docker file system" so the original ~6 min build time becomes ~25 min inside docker
the extra github action caching step totally worth it in this case